scipy/sparse/compressed.py

```diff
@@ -906,8 +918,9 @@ def _process_slice(sl, num):
 
         def _in_bounds(i0, i1, num):
             if not (0 <= i0 < num) or not (0 < i1 <= num) or not (i0 < i1):
-                raise IndexError("index out of bounds: 0<=%d<%d, 0<=%d<%d, %d<%d" %
-                                    (i0, num, i1, num, i0, i1))
+                raise IndexError("index out of bounds:"
+                                 " 0<={}<{}, 0<={}<{}, {}<{}"
+                                 "".format(i0, num, i1, num, i0, i1))
```

Comment:
This is a good candidate for named-variable formatting.

Reply:
I agree. That's a good idea.